### PR TITLE
Add option to restore automatic status

### LINKED ISSUE #29339: Missing action to revert the status back after an auto change in calendar/talk
> - In the Status modal, between the "What’s your status" input and the 5 suggestions, we put a block like this (with the second line being a text link / button:
> > Automatically set during calls and events.
> > Reset to **📅 Vacation next week**

Still missing, but not related to the Talk change

_Originally posted by @nickvergessen in https://github.com/nextcloud/spreed/issues/6110#issuecomment-946424372_

### PR DESCRIPTION
**Pull request by CarlSchwan**
Fix #29339

image

### CODE REVIEW

File: apps/user_status/src/store/backupUserStatus.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
+/**
+ * @copyright Copyright (c) 2020 Georg Ehrke
```
Comment (nickvergessen) [inline apps/user_status/src/store/backupUserStatus.js:2]: 🙈

File: apps/user_status/src/store/userStatus.js [hunk 2]
```diff
@@ -165,6 +166,31 @@ const actions = {
...
+	 * @param {Object} vuex.state The Vuex state object
+	 * @param {Object} vuex.dispatch The Vuex dispatch object
+	 * @param {Object} data The data destructuring object
+	 * @param {Object} data.status The current status
+	 * @param {Object} data.backupStatus The backup status
+	 * @returns {Promise<void>}
+	 */
+	async revertStatus({ commit, state, dispatch }, { status, backupStatus }) {
+		await revertStatus(status)
+		commit('loadStatusFromServer', backupStatus)
```
Comment (nickvergessen) [inline apps/user_status/src/store/userStatus.js:183]: Can't we return the new status on the revert API request? so we save a request and the round trip time?

File: apps/user_status/src/components/ResetStatus.vue [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+<template>
+	<div v-if="hasLoaded && backupStatus.status">
+		{{ $t('user_status', 'Automatically set during calls and events.') }}<br />
+		{{ $t('user_status', 'Reset to') }}
+		<a
+			href="#"
+			role="button"
+			@click.prevent.stop="revertCurrentStatus">
+			{{ message }}
+		</a>
```
Comment (nickvergessen) [inline apps/user_status/src/components/ResetStatus.vue:31]: As per https://docs.nextcloud.com/server/latest/developer_manual/basics/front-end/l10n.html#improving-your-translations please make the message a parameter in the translation, or use another line break and make the translated string "complete" e.g. `Reset status:` and use a real button?

File: apps/user_status/appinfo/routes.php [hunk 4]
```diff
@@ -34,6 +34,9 @@
 		['name' => 'UserStatus#setPredefinedMessage', 'url' => '/api/v1/user_status/message/predefined', 'verb' => 'PUT'],
 		['name' => 'UserStatus#setCustomMessage', 'url' => '/api/v1/user_status/message/custom', 'verb' => 'PUT'],
 		['name' => 'UserStatus#clearMessage', 'url' => '/api/v1/user_status/message', 'verb' => 'DELETE'],
+		// Routes for getting your backup status (if it exists)
+		['name' => 'UserStatus#getBackupStatus', 'url' => '/api/v1/user_status_backup', 'verb' => 'GET'],
```
Comment (nickvergessen) [inline apps/user_status/appinfo/routes.php:38]:
```suggestion
		['name' => 'UserStatus#getBackupStatus', 'url' => '/api/v1/user_status/backup', 'verb' => 'GET'],
```

To align with other endpoints?

**Comment by nickvergessen** [comment]
> Internal Server Error\nArgument 1 passed to OCA\\UserStatus\\Controller\\UserStatusController::setPredefinedMessage() must be of the type string, null given, called in server/lib/private/AppFramework/Http/Dispatcher.php on line 217 in file 'server/apps/user_status/lib/Controller/UserStatusController.php' line 133

**Comment by jancborchardt** [comment]
This note should show below the input field for the status message though, as it's related to the message. Right?

File: apps/user_status/src/store/backupUserStatus.js (hunk 1 above)
Comment (CarlSchwan) [reply apps/user_status/src/store/backupUserStatus.js:2]: Some part of this files are adapted from apps/user_status/src/store/userStatus.js

**Comment by jancborchardt** [comment]
> This note should show below the input field for the status message though, as it's related to the message. Right?

@CarlSchwan was that fixed? Also it seems this needs rebasing, mind updating the screenshot afterwards? :)

**Comment by PVince81** [comment]
did we not fix this differently ? or have plans for it in a different context ? @CarlSchwan @nickvergessen

**Comment by nickvergessen** [comment]
This would still be highly appiciated

**Review by jancborchardt — CHANGES_REQUESTED** [review]
This note should show below the input field for the status message though, as it's related to the message.

**Comment by skjnldsv** [comment]
cc @nickvergessen do we still need this?
also cc @ChristophWurst as it seems groupware also have meeting automated statuses

**Pull request closed without merging**